keras_cv/models/resnet_v2.py

```diff
@@ -86,6 +96,62 @@
 """
 
 
+def BasicBlock(filters, kernel_size=3, stride=1, conv_shortcut=False, name=None):
+    """A basic residual block (v2).
+    Args:
+        filters: integer, filters of the basic layer.
+        kernel_size: default 3, kernel size of the bottleneck layer.
+        stride: default 1, stride of the first layer.
+        conv_shortcut: default False, use convolution shortcut if True,
+          otherwise identity shortcut.
+        name: string, block label.
+    Returns:
+      Output tensor for the residual block.
+    """
+    if name is None:
+        name = f"v2_basic_block_{backend.get_uid('v2_basic_block')}"
+
+    def apply(x):
+        use_preactivation = layers.BatchNormalization(
+            axis=BN_AXIS, epsilon=1.001e-5, name=name + "_use_preactivation_bn"
+        )(x)
+
+        use_preactivation = layers.Activation(
+            "relu", name=name + "_use_preactivation_relu"
+        )(use_preactivation)
+
+        if conv_shortcut:
+            shortcut = layers.Conv2D(
+                filters, 1, strides=stride, name=name + "_0_conv"
+            )(use_preactivation)
+        else:
+            shortcut = layers.MaxPooling2D(1, strides=stride)(x) if stride > 1 else x
+
+        x = layers.Conv2D(
+            filters, kernel_size, padding="SAME", strides=1, use_bias=False, name=name + "_1_conv"
+        )(
+            use_preactivation
+        )
+        x = layers.BatchNormalization(
+            axis=BN_AXIS, epsilon=1.001e-5, name=name + "_1_bn"
+        )(x)
+        x = layers.Activation("relu", name=name + "_1_relu")(x)
+
+        x = layers.ZeroPadding2D(padding=((1, 1), (1, 1)), name=name + "_2_pad")(x)
```

Comment:
Here I am not sure I understand why we want manual padding, instead of specifying the padding type in the conv right after.